qemu/tests/nbd_unix_connection.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from virttest import data_dir
+from virttest import storage
+from virttest import qemu_storage
+from virttest import error_context
+
+from avocado.utils import process
+
+from provider.nbd_image_export import QemuNBDExportImage
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    1) Clone the system image1 with qemu-img
+    2) Export the cloned image with qemu-nbd(type=unix)
+    3) Stop to export nbd image
+    4) Repeat connect nbd image by nbdsh cmd
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _prepare():
+        logging.info("Clone system image with qemu-img")
+        result = qemu_storage.QemuImg(
+            params, None, params['images'].split()[0]).dd(
+                output=storage.get_image_filename(
+                    params.object_params(params["local_image_tag"]),
+                    data_dir.get_data_dir()
+                ),
+                bs=1024*1024
+        )
+        if result.exit_status != 0:
+            test.fail('Failed to clone the system image, error: %s'
+                      % result.stderr.decode())
+
+    def run_nbd_connect_cmd(cmd):
+        result = process.run(cmd, timeout=5, ignore_status=True, shell=True)
+        if result.exit_status not in {0, -15}:
+            test.fail('Failed to connect nbd by unix socket, error: %s'
+                      % result.stderr.decode())
+
+    _prepare()
+
+    # local image to be exported
+    nbd_export = QemuNBDExportImage(params, params["local_image_tag"])
+    nbd_export.export_image()
+    # stop/suspend export
+    logging.info("Stop export via qemu-nbd:")
```

Comment:
you may move the logging into the suspend function above, of course, it's OK to leave it here, but I suggest to change the msg to 'Suspend qemu-nbd to stop export'

Reply:
Thanks, refined the log.